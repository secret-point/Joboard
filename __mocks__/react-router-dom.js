// Mock navigate and useParams when needed
// const navigate = jest.fn();
// const useParams = jest.fn();
const useLocation = jest.fn();

module.exports = {
  ...jest.requireActual("react-router-dom"),
  // useNavigate: () => navigate,
  // useParams,
  useLocation,
};
