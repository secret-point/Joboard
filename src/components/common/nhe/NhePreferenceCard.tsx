import React from "react";
import { Col, Row } from "@amzn/stencil-react-components/layout";
import { H4, Text } from "@amzn/stencil-react-components/text";
import { IconArrowRight, IconSize } from "@amzn/stencil-react-components/icons";
import { translate as t } from "../../../utils/translator";
import { routeToAppPageWithPath } from "../../../utils/helper";
import { PAGE_ROUTES } from "../../pageRoutes";

const NhePreferenceCard = () => {
  return (
    <Col
      className="nhePreferenceCard"
      padding="S300"
      gridGap={8}
    >
      <H4>
        {t("BB-nhe-preference-card-title-text", "Don't see an option that works for you?")}
      </H4>
      <Text fontSize="T200">
        {t("BB-nhe-preference-card-content-text", "You can submit your event preferences and we'll contact you when we find a time that works for you.")}
      </Text>
      <Row padding={{ top: "S400" }}>
        <Row
          className="nhePreferenceLink"
          gridGap={8}
          onClick={() => {
            routeToAppPageWithPath(PAGE_ROUTES.NHE_PREFERENCES);
          }}
        >
          <Text fontWeight="medium" fontSize="T200">
            {t("BB-nhe-preference-card-go-to-nhe-preference-link", "Select your pre-hire event preferences")}
          </Text>
          <Row padding={{ top: "S100" }}>
            <IconArrowRight size={IconSize.Medium} fontSize="T100" aria-hidden />
          </Row>
        </Row>
      </Row>
    </Col>
  );
};

export default NhePreferenceCard;