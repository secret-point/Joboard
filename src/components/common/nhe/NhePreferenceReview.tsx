import React from "react";
import { Col } from "@amzn/stencil-react-components/layout";
import { Card } from "@amzn/stencil-react-components/card";
import { Text } from "@amzn/stencil-react-components/text";
import { NHEPreferences } from "../../../utils/types/common";
import { translate as t } from "../../../utils/translator";

interface NhePreferenceReviewProps {
  nhePreference: NHEPreferences;
}

export const NhePreferenceReview = (props: NhePreferenceReviewProps) => {
  const { nhePreference } = props;
  const { locations, preferredNHEDates, preferredNHETimeIntervals } = nhePreference;

  return (
    <Card width="100%" padding="S300" isElevated>
      <Col width="100%" padding="S300" gridGap={15}>
        <Text fontSize="T400" fontWeight="bold">
          {t("BB-nhe-preference-review-title-text", "Pre-hire appointment preferences")}
        </Text>
        <Text fontWeight="bold">
          {t("BB-nhe-preference-review-preferred-city-text", "Preferred city")}
        </Text>
        <ul className="ul-list">
          {
            locations?.map(location => (
              <li key={location}><Text fontSize="T200">{location}</Text></li>
            ))
          }
        </ul>
        <Text fontWeight="bold">
          {t("BB-nhe-preference-review-preferred-dates-text", "Preferred date(s)")}
        </Text>
        <ul className="ul-list">
          {
            preferredNHEDates?.map(date => (
              <li key={date}><Text fontSize="T200">{date}</Text></li>
            ))
          }
        </ul>
        <Text fontWeight="bold">
          {t("BB-nhe-preference-review-preferred-times-text", "Preferred time(s)")}
        </Text>
        <ul className="ul-list">
          {
            preferredNHETimeIntervals?.map(time => (
              <li key={time}><Text fontSize="T200">{time}</Text></li>
            ))
          }
        </ul>
      </Col>
    </Card>
  );
};

export default NhePreferenceReview;
