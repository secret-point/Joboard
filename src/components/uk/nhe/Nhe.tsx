import React, { useEffect, useState } from "react";
import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { Col, Row } from "@amzn/stencil-react-components/layout";
import { H4, Label, Text } from "@amzn/stencil-react-components/text";
import queryString from "query-string";
import { connect } from "react-redux";
import { useLocation } from "react-router-dom";
import { boundResetBannerMessage } from "../../../actions/UiActions/boundUi";
import { log } from "../../../helpers/log-helper";
import {
  getPageNameFromPath,
  parseQueryParamsArrayToSingleItem,
  resetIsPageMetricsUpdated
} from "../../../helpers/utils";
import { ApplicationState } from "../../../reducers/application.reducer";
import { CandidateState } from "../../../reducers/candidate.reducer";
import { JobState } from "../../../reducers/job.reducer";
import { NheState } from "../../../reducers/nhe.reducer";
import { ScheduleState } from "../../../reducers/schedule.reducer";
import { ApplicationStepList } from "../../../utils/constants/common";
import {
  checkAndBoundGetApplication,
  getDefaultUkNheApptTimeFromMap,
  getLocale,
  getUKNHEAppointmentTimeMap,
  getUKNHEAppointmentTitleList,
  getUKNHEMaxSlotLength,
  handleConfirmUKNHESelection,
  initiateScheduleDetailOnPageLoad
} from "../../../utils/helper";
import { translate as t } from "../../../utils/translator";
import { GetNheTimeSlotRequestThroughNheDS, NHETimeSlotUK, ScheduleUK } from "../../../utils/types/common";
import StepHeader from "../../common/StepHeader";
import DebouncedButton from "../../common/DebouncedButton";
import { Select } from "@amzn/stencil-react-components/form";
import { Popover } from "@amzn/stencil-react-components/popover";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import { boundGetCandidateInfo } from "../../../actions/CandidateActions/boundCandidateActions";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { boundGetNheTimeSlotsThroughNheDs } from "../../../actions/NheActions/boundNheAction";
import { boundGetScheduleDetail } from "../../../actions/ScheduleActions/boundScheduleActions";
import moment from "moment";
import NhePreferenceCard from "../../common/nhe/NhePreferenceCard";
import { NHE_SLOTS_TO_DISPLAY_NHE_PREFERENCES } from "../../../utils/enums/common";
import { PAGE_ROUTES } from "../../pageRoutes";

interface MapStateToProps {
  job: JobState;
  application: ApplicationState;
  schedule: ScheduleState;
  candidate: CandidateState;
  nhe: NheState;
}

export const Nhe = ( props: MapStateToProps ) => {
  const { job, application, schedule, candidate, nhe } = props;
  const { search, pathname } = useLocation();
  const pageName = getPageNameFromPath(pathname);
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const { applicationId, jobId, scheduleId } = queryParams;
  const jobDetail = job.results;
  const applicationData = application.results;
  const scheduleDetail = schedule.results.scheduleDetail as ScheduleUK;
  const candidateData = candidate.results?.candidateData;
  const nheData = nhe.results.nheData as NHETimeSlotUK[];
  const showNhePreferenceCard = nheData.length <= NHE_SLOTS_TO_DISPLAY_NHE_PREFERENCES;
  const appointmentDateList = getUKNHEAppointmentTitleList(nheData);
  const appointmentTimeMap = getUKNHEAppointmentTimeMap(nheData);

  const [selectedNheDate, setSelectedNheDate] = useState<string>("");
  const [selectedNheTime, setSelectedNheTime] = useState<string>("");
  const maxNheSlotLength = nheData.length > 0 ? getUKNHEMaxSlotLength(nheData, scheduleId) : 0;

  const displayFirstName = candidateData?.preferredFirstName || candidateData?.firstName || "";
  const displayLastName = candidateData?.lastName || "";

  useEffect(() => {
    // Refresh and add scheduleId in the url from the jobSelected if it doesn't exist from the query param
    if (!scheduleId && applicationData) {
      initiateScheduleDetailOnPageLoad(applicationData, PAGE_ROUTES.NHE);
    }
  }, [applicationData]);

  useEffect(() => {
    const defaultAptTime = getDefaultUkNheApptTimeFromMap(appointmentTimeMap, appointmentDateList[0]);
    setSelectedNheDate(appointmentDateList[0]);
    setSelectedNheTime(defaultAptTime[0]);
  }, [nheData]);

  useEffect(() => {
    applicationId && checkAndBoundGetApplication(applicationId);
  }, [applicationId]);

  useEffect(() => {
    scheduleDetail &&jobDetail && applicationData && nheData.length && addMetricForPageLoad(pageName);

  }, [jobDetail, applicationData, scheduleDetail, nheData, pageName]);

  useEffect(() => {
    boundGetCandidateInfo();
  }, []);

  // Don't refetch data if id is not changing
  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() });
  }, [jobDetail, jobId]);

  useEffect(() => {
    scheduleId && scheduleId!== scheduleDetail?.scheduleId && boundGetScheduleDetail({
      locale: getLocale(),
      scheduleId: scheduleId
    });
  }, [scheduleDetail, scheduleId]);

  useEffect(() => {
    // only use call NHE through requisition service, others call nhe service directly
    if (scheduleDetail) {
      const request: GetNheTimeSlotRequestThroughNheDS = {
        siteId: scheduleDetail?.siteId,
        startDate: new Date().toISOString().split("T")[0],
        locale: getLocale(),
        endDate: moment(scheduleDetail?.hireStartDate, "YYYY-MM-DD")
          .subtract(scheduleDetail?.contingencyTat, "days")
          .format("YYYY-MM-DD")
      };

      // We don't redirect to no Nhe found when there is no NHE
      boundGetNheTimeSlotsThroughNheDs(request, false);
    }
  }, [scheduleDetail]);

  useEffect(() => {
    return () => {
      // reset this so as it can emit new pageload event after being unmounted.
      resetIsPageMetricsUpdated(pageName);
    };
  }, [pageName]);

  const handleConfirmSelection = () => {
    boundResetBannerMessage();
    const selectedNhe: NHETimeSlotUK = nheData.filter(nhe => nhe.title === selectedNheDate && nhe.startTime === selectedNheTime)[0];

    if (applicationData && selectedNhe) {
      log("Handing handleConfirmNHESelection: ", {
        application: applicationData,
        selectedNhe: selectedNhe
      });
      handleConfirmUKNHESelection(applicationData, selectedNhe as NHETimeSlotUK);
    }
  };

  return (
    <Col className="pageContainerWithMarginTop" margin={{ top: "S300" }}>
      <StepHeader jobTitle={jobDetail?.jobTitle || ""} step={ApplicationStepList[2]} />
      <Col padding={{ top: "S400" }} gridGap={20}>
        <Col gridGap={10}>
          <H4>
            {t("BB-nhe-page-header-text", "Schedule pre-hire appointment")}
          </H4>
          <Text fontSize="T200">
            {t("BB-kondo-nhe-page-title-text", `You are almost there, ${displayFirstName} ${displayLastName}! Select your preferred date and time to schedule a pre-hire appointment where we check your work authorisation documents.`)}
          </Text>
        </Col>
        <Row>
          <Popover
            triggerText={t("BB-Kondo-pre-hire-appointment-description-popover-title", "Pre-hire appointment is a virtual event")}
            shouldCloseOnFocusLeave={false}
          >
            {({ close }) => (
              <Col gridGap="S500" alignItems="flex-start">
                <Text>
                  {t("BB-Kondo-pre-hire-appointment-description-popover-content", "Pre - hire appointment will be held over video conference. Once you choose a date and time to connect with us, we will email you appointment details and link. If you have any questions or technical issues, please contact us.")}
                </Text>
                <Button onClick={close} variant={ButtonVariant.Tertiary}>
                  {t("BB-Kondo-pre-hire-appointment-description-popover-close-button", "Close")}
                </Button>
              </Col>
            )}
          </Popover>
        </Row>
        {nheData?.length > 0 && (
          <Col>
            <Col gridGap="S400">
              <Col gridGap={12}>
                <Label htmlFor="nheDateSelect" id="nheDateLabel">
                  {t("BB-Kondo-nhe-appointment-input-date-label", "Choose appointment date")} *
                </Label>
                <Select
                  id="nheDateSelect"
                  onChange={(option) => {
                    setSelectedNheDate(option);
                    setSelectedNheTime(getDefaultUkNheApptTimeFromMap(appointmentTimeMap, option)[0]);
                  }}
                  options={appointmentDateList}
                  defaultValue={appointmentDateList[0]}
                  value={selectedNheDate}
                />
              </Col>

              <Col gridGap={12}>
                <Label htmlFor="nheTimeSelect" id="nheTimeLabel">
                  {t("BB-Kondo-nhe-appointment-time-input-label", "Choose appointment time")} *
                </Label>
                <Select
                  id="nheTimeSelect"
                  onChange={(option) => {
                    setSelectedNheTime(option);
                  }}
                  options={appointmentTimeMap.get(selectedNheDate) || []}
                  defaultValue={getDefaultUkNheApptTimeFromMap(appointmentTimeMap, selectedNheDate)[0]}
                  value={selectedNheTime}
                />
              </Col>

              {
                !!maxNheSlotLength && (
                  <Text fontSize="T100">
                    {t("BB-kondo-nhe-nhe-slot-duration-notice", `All appointments slots are displayed in UK local time. Each appointment is up to ${maxNheSlotLength} minutes long.`, { maxNheSlotLength })}
                  </Text>
                )
              }

            </Col>
            <Col className="nhe-sticky-button" padding={{ top: "S300" }}>
              <DebouncedButton
                variant={ButtonVariant.Primary}
                onClick={() => {
                  handleConfirmSelection();
                }}
                debounceTime={1000}
              >
                {t("BB-kondo-nhe-page-confirm-selection-button-text", "Confirm Selection")}
              </DebouncedButton>
            </Col>
          </Col>
        )}
        {
          showNhePreferenceCard && <NhePreferenceCard />
        }
      </Col>
    </Col>
  );
};

const mapStateToProps = ( state: MapStateToProps ) => {
  return state;
};

export default connect(mapStateToProps)(Nhe);
