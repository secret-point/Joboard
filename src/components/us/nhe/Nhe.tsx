import React, { useEffect, useState } from "react";
import { ButtonVariant } from "@amzn/stencil-react-components/button";
import { Col } from "@amzn/stencil-react-components/layout";
import { H4, Text } from "@amzn/stencil-react-components/text";
import queryString from "query-string";
import { connect } from "react-redux";
import { useLocation } from "react-router-dom";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import { boundGetCandidateInfo } from "../../../actions/CandidateActions/boundCandidateActions";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { boundGetScheduleDetail } from "../../../actions/ScheduleActions/boundScheduleActions";
import { boundResetBannerMessage } from "../../../actions/UiActions/boundUi";
import { log } from "../../../helpers/log-helper";
import {
  getPageNameFromPath,
  parseQueryParamsArrayToSingleItem,
  resetIsPageMetricsUpdated
} from "../../../helpers/utils";
import { ApplicationState } from "../../../reducers/application.reducer";
import { CandidateState } from "../../../reducers/candidate.reducer";
import { JobState } from "../../../reducers/job.reducer";
import { NheState } from "../../../reducers/nhe.reducer";
import { ScheduleState } from "../../../reducers/schedule.reducer";
import { ApplicationStepList, localeToLanguageMap } from "../../../utils/constants/common";
import {
  checkAndBoundGetApplication,
  fetchNheTimeSlotDs,
  getLocale,
  handleConfirmNHESelection,
  nheGroupByLanguage
} from "../../../utils/helper";
import { translate as t } from "../../../utils/translator";
import { NHETimeSlot } from "../../../utils/types/common";
import NheTimeSlotCard from "../../common/nhe/NheTimeSlotCard";
import StepHeader from "../../common/StepHeader";
import DebouncedButton from "../../common/DebouncedButton";

interface MapStateToProps {
  job: JobState;
  application: ApplicationState;
  schedule: ScheduleState;
  candidate: CandidateState;
  nhe: NheState;
}

interface JobOpportunityProps {

}

type JobOpportunityMergedProps = MapStateToProps & JobOpportunityProps;

export const Nhe = ( props: JobOpportunityMergedProps ) => {
  const { job, application, schedule, candidate, nhe } = props;
  const { search, pathname } = useLocation();
  const pageName = getPageNameFromPath(pathname);
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const { applicationId, jobId, scheduleId } = queryParams;
  const jobDetail = job.results;
  const applicationData = application.results;
  const { scheduleDetail } = schedule.results;
  const candidateData = candidate.results?.candidateData;
  const { nheData } = nhe.results;
  const nheGroups = nheGroupByLanguage(nheData);
  const nheGroupsCurrentLanguageCount = nheGroups.get("current")?.length || 0;
  const nheGroupsOtherLanguageCount = nheGroups.get("other")?.length || 0;
  const [selectedNhe, setSelectedNhe] = useState<NHETimeSlot>();

  useEffect(() => {
    applicationId && checkAndBoundGetApplication(applicationId);
  }, [applicationId]);

  useEffect(() => {
    scheduleDetail &&jobDetail && applicationData && nheData.length && addMetricForPageLoad(pageName);

  }, [jobDetail, applicationData, scheduleDetail, nheData, pageName]);

  useEffect(() => {
    boundGetCandidateInfo();
  }, []);

  // Don't refetch data if id is not changing
  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() });
  }, [jobDetail, jobId]);

  useEffect(() => {
    scheduleId && scheduleId!== scheduleDetail?.scheduleId && boundGetScheduleDetail({
      locale: getLocale(),
      scheduleId: scheduleId
    });
  }, [scheduleDetail, scheduleId]);

  useEffect(() => {
    // only use call NHE through requisition service, others call nhe service directly
    scheduleDetail && fetchNheTimeSlotDs(scheduleDetail, applicationId, false);
  }, [scheduleDetail]);

  useEffect(() => {
    return () => {
      // reset this so as it can emit new pageload event after being unmounted.
      resetIsPageMetricsUpdated(pageName);
    };
  }, [pageName]);

  const handleConfirmSelection = () => {
    boundResetBannerMessage();

    if (applicationData && selectedNhe) {
      log("Handing handleConfirmNHESelection: ", {
        application: applicationData,
        selectedNhe: selectedNhe
      });
      handleConfirmNHESelection(applicationData, selectedNhe);
    }
  };

  const displayFirstName = candidateData?.preferredFirstName || candidateData?.firstName || "";
  const displayLastName = candidateData?.lastName || "";

  return (
    <Col id="jobOpportunityContainer">
      {
        // TODO need to align how each application workflow is related to app steps
      }
      <StepHeader jobTitle={jobDetail?.jobTitle || ""} step={ApplicationStepList[2]} />
      <Col padding={{ top: "S400" }} gridGap={20}>
        <Col gridGap={10}>
          <H4>
            {t("BB-nhe-page-header-text", "Schedule pre-hire appointment")}
          </H4>
          <Text fontSize="T200">
            {t("BB-nhe-page-title-text", `You are almost there, ${displayFirstName} ${displayLastName}! We need you to come for a badge photo, complete work authorization and a drug test, if applicable.`, { firstname: displayFirstName, lastname: displayLastName })}
          </Text>
        </Col>
        <Col padding={{ top: "S400" }} gridGap={15}>
          {
            nheGroupsCurrentLanguageCount > 0 && nheGroupsOtherLanguageCount > 0 && (
              <H4 className="nheGroupingHeader">
                {`${t(`BB-nhe-time-slots-available-in-language-${nheGroupsCurrentLanguageCount === 1 ? "singular" : "plural"}`, `${nheGroupsCurrentLanguageCount} time slot${nheGroupsCurrentLanguageCount === 1 ? "" : "s"} available in ${t(...localeToLanguageMap[getLocale()])}`, { numberOfAppointments: nheGroupsCurrentLanguageCount, currentLanguage: t(...localeToLanguageMap[getLocale()]) })}`}
              </H4>
            )
          }
          {
            nheGroups.get("current")?.map(nheItem => (
              <NheTimeSlotCard
                nheTimeSlot={nheItem} key={nheItem.timeSlotId}
                handleChange={(nheSlotTime: NHETimeSlot) => setSelectedNhe(nheSlotTime)}
              />
            ))
          }
          {
            nheGroupsCurrentLanguageCount > 0 && nheGroupsOtherLanguageCount > 0 && (
              <H4 className="nheGroupingHeader">
                {`${t(`BB-nhe-time-slots-in-other-languages-${nheGroupsOtherLanguageCount === 1 ? "singular" : "plural"}`, `${nheGroupsOtherLanguageCount} time slot${nheGroupsOtherLanguageCount === 1 ? "" : "s"} available in other languages`, { numberOfAppointments: nheGroupsOtherLanguageCount })}`}
              </H4>
            )
          }
          {
            nheGroups.get("other")?.map(nheItem => (
              <NheTimeSlotCard
                nheTimeSlot={nheItem} key={nheItem.timeSlotId}
                handleChange={(nheSlotTime: NHETimeSlot) => setSelectedNhe(nheSlotTime)}
              />
            ))
          }
        </Col>
        { selectedNhe && (
          <Col className="nhe-sticky-button">
            <DebouncedButton
              variant={ButtonVariant.Primary}
              onClick={() => {
                handleConfirmSelection();
              }}
              debounceTime={1000}
            >
              {t("BB-nhe-page-confirm-selection-button-text", "Confirm Selection")}
            </DebouncedButton>
          </Col>
        ) }
      </Col>
    </Col>
  );
};

const mapStateToProps = ( state: MapStateToProps ) => {
  return state;
};

export default connect(mapStateToProps)(Nhe);
