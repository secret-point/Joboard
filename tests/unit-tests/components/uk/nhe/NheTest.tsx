import React from "react";
import { shallow } from "enzyme";
import { useLocation } from "react-router-dom";
import { Nhe } from "../../../../../src/components/uk/nhe/Nhe";
import {
  TEST_APPLICATION_ID,
  TEST_APPLICATION_STATE,
  TEST_CANDIDATE_STATE,
  TEST_JOB_ID,
  TEST_JOB_STATE,
  TEST_NHE_DATA_UK,
  TEST_NHE_STATE,
  TEST_SCHEDULE_ID,
  TEST_SCHEDULE_STATE,
  TestPossibleNheDates,
  TestSavePossibleNheDateRequest
} from "../../../../test-utils/test-data";

describe("Nhe", () => {
  const mockLocation = {
    pathname: "/nhe",
    search: `?jobId=${TEST_JOB_ID}&applicationId=${TEST_APPLICATION_ID}&scheduleId=${TEST_SCHEDULE_ID}`,
    hash: "",
    state: null
  };
  const mockUseLocation = useLocation as jest.Mock;
  mockUseLocation.mockReturnValue(mockLocation);

  it("should match snapshot with NHE Preference Card", () => {
    const shallowWrapper = shallow(
      <Nhe
        candidate={TEST_CANDIDATE_STATE}
        job={TEST_JOB_STATE}
        application={TEST_APPLICATION_STATE}
        schedule={TEST_SCHEDULE_STATE}
        nhe={{
          ...TEST_NHE_STATE,
          results: {
            nheData: TEST_NHE_DATA_UK,
            possibleNhePreferences: TestPossibleNheDates
          },
          nhePreferenceRequest: TestSavePossibleNheDateRequest
        }
        }
      />);

    expect(shallowWrapper).toMatchSnapshot();
  });
  it("should match snapshot without NHE Preference Card", () => {
    const shallowWrapper = shallow(
      <Nhe
        candidate={TEST_CANDIDATE_STATE}
        job={TEST_JOB_STATE}
        application={TEST_APPLICATION_STATE}
        schedule={TEST_SCHEDULE_STATE}
        nhe={{
          ...TEST_NHE_STATE,
          results: {
            nheData: TEST_NHE_DATA_UK.splice(0, 2),
            possibleNhePreferences: TestPossibleNheDates
          },
          nhePreferenceRequest: TestSavePossibleNheDateRequest
        }
        }
      />);

    expect(shallowWrapper).toMatchSnapshot();
  });
});
