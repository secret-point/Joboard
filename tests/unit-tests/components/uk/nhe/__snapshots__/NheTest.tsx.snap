// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nhe should match snapshot with NHE Preference Card 1`] = `
<Col
  className="pageContainerWithMarginTop"
  margin={
    Object {
      "top": "S300",
    }
  }
>
  <StepHeader
    jobTitle=""
    step={
      Object {
        "title": "Schedule pre-hire appointment",
        "titleTranslationKey": "BB-ApplicationSteps-schedule-preHire-appt-text",
      }
    }
    steps={
      Array [
        Object {
          "title": "Select Job",
          "titleTranslationKey": "BB-ApplicationSteps-select-job-text",
        },
        Object {
          "title": "Enter required information",
          "titleTranslationKey": "BB-ApplicationSteps-enter-required-information-text",
        },
        Object {
          "title": "Schedule pre-hire appointment",
          "titleTranslationKey": "BB-ApplicationSteps-schedule-preHire-appt-text",
        },
      ]
    }
  />
  <Col
    gridGap={20}
    padding={
      Object {
        "top": "S400",
      }
    }
  >
    <Col
      gridGap={10}
    >
      <H4
        color="neutral90"
        font="primary"
        fontSize="T400"
        fontWeight="regular"
        hyphens="none"
        margin={0}
      >
        Schedule pre-hire appointment
      </H4>
      <Text
        font="primary"
        fontSize="T200"
        hyphens="none"
      >
        You are almost there, test-firstname test-lastname! Select your preferred date and time to schedule a pre-hire appointment where we check your work authorisation documents.
      </Text>
    </Col>
    <Row>
      <Popover
        position="bottom-leading"
        shouldCloseOnFocusLeave={false}
        shouldFocusOnOpen={false}
        triggerButtonProps={Object {}}
        triggerText="Pre-hire appointment is a virtual event"
      >
        <Component />
      </Popover>
    </Row>
    <Col>
      <Col
        gridGap="S400"
      >
        <Col
          gridGap={12}
        >
          <Label
            color="neutral90"
            font="primary"
            fontSize="T200"
            fontWeight="regular"
            htmlFor="nheDateSelect"
            hyphens="none"
            id="nheDateLabel"
          >
            Choose appointment date
             *
          </Label>
          <Select
            collapsedOptionsExpanderTitle="Selected items"
            customSelectionOptionsLabel="Group selection options"
            defaultOptionsLabel="Single selection options"
            defaultValue="Friday, Jan 20th 2023"
            disabledTagsHiddenText="The options in this list are not dismissible because the input is disabled"
            getTriggerText={[Function]}
            id="nheDateSelect"
            layout="default"
            noOptionsText="No options available"
            onChange={[Function]}
            options={
              Array [
                "Friday, Jan 20th 2023",
                "Saturday, Jan 21st 2023",
              ]
            }
            placeholder="Choose an option"
            searchFieldAriaLabel="Start typing to filter out search results"
            selectedOptionAltText="selected"
            value=""
            valueAccessor="value"
            width="100%"
          />
        </Col>
        <Col
          gridGap={12}
        >
          <Label
            color="neutral90"
            font="primary"
            fontSize="T200"
            fontWeight="regular"
            htmlFor="nheTimeSelect"
            hyphens="none"
            id="nheTimeLabel"
          >
            Choose appointment time
             *
          </Label>
          <Select
            collapsedOptionsExpanderTitle="Selected items"
            customSelectionOptionsLabel="Group selection options"
            defaultOptionsLabel="Single selection options"
            disabledTagsHiddenText="The options in this list are not dismissible because the input is disabled"
            getTriggerText={[Function]}
            id="nheTimeSelect"
            layout="default"
            noOptionsText="No options available"
            onChange={[Function]}
            options={Array []}
            placeholder="Choose an option"
            searchFieldAriaLabel="Start typing to filter out search results"
            selectedOptionAltText="selected"
            value=""
            valueAccessor="value"
            width="100%"
          />
        </Col>
        <Text
          font="primary"
          fontSize="T100"
          hyphens="none"
        >
          All appointments slots are displayed in UK local time. Each appointment is up to 180 minutes long.
        </Text>
      </Col>
      <Col
        className="nhe-sticky-button"
        padding={
          Object {
            "top": "S300",
          }
        }
      >
        <DebouncedButton
          debounceTime={1000}
          onClick={[Function]}
          variant="primary"
        >
          Confirm Selection
        </DebouncedButton>
      </Col>
    </Col>
  </Col>
</Col>
`;

exports[`Nhe should match snapshot without NHE Preference Card 1`] = `
<Col
  className="pageContainerWithMarginTop"
  margin={
    Object {
      "top": "S300",
    }
  }
>
  <StepHeader
    jobTitle=""
    step={
      Object {
        "title": "Schedule pre-hire appointment",
        "titleTranslationKey": "BB-ApplicationSteps-schedule-preHire-appt-text",
      }
    }
    steps={
      Array [
        Object {
          "title": "Select Job",
          "titleTranslationKey": "BB-ApplicationSteps-select-job-text",
        },
        Object {
          "title": "Enter required information",
          "titleTranslationKey": "BB-ApplicationSteps-enter-required-information-text",
        },
        Object {
          "title": "Schedule pre-hire appointment",
          "titleTranslationKey": "BB-ApplicationSteps-schedule-preHire-appt-text",
        },
      ]
    }
  />
  <Col
    gridGap={20}
    padding={
      Object {
        "top": "S400",
      }
    }
  >
    <Col
      gridGap={10}
    >
      <H4
        color="neutral90"
        font="primary"
        fontSize="T400"
        fontWeight="regular"
        hyphens="none"
        margin={0}
      >
        Schedule pre-hire appointment
      </H4>
      <Text
        font="primary"
        fontSize="T200"
        hyphens="none"
      >
        You are almost there, test-firstname test-lastname! Select your preferred date and time to schedule a pre-hire appointment where we check your work authorisation documents.
      </Text>
    </Col>
    <Row>
      <Popover
        position="bottom-leading"
        shouldCloseOnFocusLeave={false}
        shouldFocusOnOpen={false}
        triggerButtonProps={Object {}}
        triggerText="Pre-hire appointment is a virtual event"
      >
        <Component />
      </Popover>
    </Row>
    <Col>
      <Col
        gridGap="S400"
      >
        <Col
          gridGap={12}
        >
          <Label
            color="neutral90"
            font="primary"
            fontSize="T200"
            fontWeight="regular"
            htmlFor="nheDateSelect"
            hyphens="none"
            id="nheDateLabel"
          >
            Choose appointment date
             *
          </Label>
          <Select
            collapsedOptionsExpanderTitle="Selected items"
            customSelectionOptionsLabel="Group selection options"
            defaultOptionsLabel="Single selection options"
            defaultValue="Friday, Jan 20th 2023"
            disabledTagsHiddenText="The options in this list are not dismissible because the input is disabled"
            getTriggerText={[Function]}
            id="nheDateSelect"
            layout="default"
            noOptionsText="No options available"
            onChange={[Function]}
            options={
              Array [
                "Friday, Jan 20th 2023",
              ]
            }
            placeholder="Choose an option"
            searchFieldAriaLabel="Start typing to filter out search results"
            selectedOptionAltText="selected"
            value=""
            valueAccessor="value"
            width="100%"
          />
        </Col>
        <Col
          gridGap={12}
        >
          <Label
            color="neutral90"
            font="primary"
            fontSize="T200"
            fontWeight="regular"
            htmlFor="nheTimeSelect"
            hyphens="none"
            id="nheTimeLabel"
          >
            Choose appointment time
             *
          </Label>
          <Select
            collapsedOptionsExpanderTitle="Selected items"
            customSelectionOptionsLabel="Group selection options"
            defaultOptionsLabel="Single selection options"
            disabledTagsHiddenText="The options in this list are not dismissible because the input is disabled"
            getTriggerText={[Function]}
            id="nheTimeSelect"
            layout="default"
            noOptionsText="No options available"
            onChange={[Function]}
            options={Array []}
            placeholder="Choose an option"
            searchFieldAriaLabel="Start typing to filter out search results"
            selectedOptionAltText="selected"
            value=""
            valueAccessor="value"
            width="100%"
          />
        </Col>
        <Text
          font="primary"
          fontSize="T100"
          hyphens="none"
        >
          All appointments slots are displayed in UK local time. Each appointment is up to 180 minutes long.
        </Text>
      </Col>
      <Col
        className="nhe-sticky-button"
        padding={
          Object {
            "top": "S300",
          }
        }
      >
        <DebouncedButton
          debounceTime={1000}
          onClick={[Function]}
          variant="primary"
        >
          Confirm Selection
        </DebouncedButton>
      </Col>
    </Col>
    <NhePreferenceCard />
  </Col>
</Col>
`;
